run_tests.sh

```diff
@@ -20,19 +20,52 @@ set -ex
 GOVERSION=${1:-1.9}
 REPO=dcrd
 
-TESTDIRS=$(go list ./... | grep -v '/vendor/')
-TESTCMD="test -z \"\$(gometalinter --vendor --disable-all \
-  --enable=gofmt \
-  --enable=vet \
-  --enable=unconvert \
-  --deadline=10m ./... 2>&1 | tee /dev/stderr)\"&& \
-  env GORACE='halt_on_error=1' go test -short -race \
-  -tags rpctest \
-  \${TESTDIRS}"
+testrepo () {
+  TESTDIRS=$(go list ./... | grep -v '/vendor/')
+  TMPFILE=$(mktemp)
+
+  # check lockfile
+  cp Gopkg.lock $TMPFILE && dep ensure && diff Gopkg.lock $TMPFILE >/dev/null
+  if [ $? != 0 ]; then
+    echo 'lockfile must be updated with dep ensure'
+    exit 1
+  fi
+
+  # check linters
+  gometalinter --vendor --disable-all --deadline=10m \
+    --enable=gofmt \
+    --enable=vet \
+    --enable=unconvert ./... | tee /dev/stderr
+  if [ $? != 0 ]; then
+    echo 'gometalinter has some complaints'
+    exit 1
+  fi
+
+  # Skip binary install for local
+  if [ $GOVERSION != "local" ]; then
+    go install . ./cmd/...
+    if [ $? != 0 ]; then
+      echo 'go install failed'
+      exit 1
+    fi
+  fi
+
+  # check go tests
+  env GORACE='halt_on_error=1' go test -short -race -tags rpctest $TESTDIRS
+  if [ $? != 0 ]; then
+    echo 'go tests failed'
+    exit 1
+  fi
+}
+
+if [ $GOVERSION == "test" ]; then
```

Comment:
why have both test and local?  If we are testing locally I would assume that gometalinter and all required linters are installed

Reply:
was trying to just keep existing the same.

Reply:
i think this would be a good case where we can change the functionality.  I don't want local tests to be making network connections with 'go get' and i don't see any mention that it does this in the readme.

We can remove the "test" command and make "local" do what "test" currently does, and then docker would call the "local" command.